src/Core/Grid/Definition/Factory/AuthorizedApplicationGridDefinitionFactory.php

```diff
@@ -0,0 +1,139 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\LinkRowAction;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\ApiAccessesStatesColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\LinkColumn;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+
+/**
+ * Class AuthorizedApplicationGridDefinitionFactory is responsible for creating new instance of Authorized Application grid definition.
+ */
+final class AuthorizedApplicationGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    use DeleteActionTrait;
+
+    public const GRID_ID = 'authorized_application';
+
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher
+    ) {
+        parent::__construct($hookDispatcher);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('List of applications', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add(
+                (new LinkColumn('name'))
+                    ->setName($this->trans('Name', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'name',
+                        'route' => 'admin_authorized_applications_view',
+                        'route_param_name' => 'authorizedApplicationId',
+                        'route_param_field' => 'id_authorized_application',
+                    ])
+            )
+            ->add(
+                (new ApiAccessesStatesColumn('api_accesses_states'))
```

Comment:
I think a `ToggleColumn` whould be a better fit from what I see on the figma. The way it's done right now, you can't define the label, it's "On/Off" but with a little refacto I think you can make it customizable

Reply:
My bad, I was thinking about another column for another page!